src/components/Login/LoginContainer.tsx

```diff
@@ -91,11 +99,61 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     setAllProducts
   } = useProductContext();
   const { logout } = useLogout();
+  const lastResendWarningMessageRef = useRef("");
+  const { showAlert } = useContext(AlertModalContext);
 
   const resetStage = (): void => {
     setLoginStage("SCAN");
   };
 
+  const getResendConfirmationIfNeeded = async (): Promise<boolean> => {
+    return new Promise(resolve => {
+      if (!lastResendWarningMessageRef.current) {
+        resolve(true);
+      } else {
+        showAlert({
+          ...defaultConfirmationProps,
+          title: "Resend OTP?",
+          description: lastResendWarningMessageRef.current,
+          buttonTexts: {
+            primaryActionText: "Resend",
+            secondaryActionText: "No"
+          },
+          onOk: () => resolve(true),
+          onCancel: () => resolve(false),
+          visible: true
+        });
+      }
+    });
+  };
+
+  const handleRequestOTP = async (
+    fullNumber = mobileNumber
+  ): Promise<boolean> => {
+    try {
+      if (!(await getResendConfirmationIfNeeded())) return false;
+      const response = await requestOTP(fullNumber, codeKey, endpointTemp);
+      lastResendWarningMessageRef.current = response.warning ?? "";
+      return true;
+    } catch (e) {
+      if (e instanceof LoginLockedError) {
+        showAlert({
+          ...disabledAccessAlertProps,
+          description: e.message,
+          onOk: () => resetStage()
+        });
+      } else if (e instanceof LoginError) {
+        showAlert({
+          ...invalidEntryAlertProps,
+          description: e.message
+        });
```

Comment:
What is the default CTA for this? Do we want to reset the stage?

Reply:
hmm i think in most cases, a `LoginError` is thrown here when the user is using an already used QR code, or a QR code that is valid but not inside our database. it does make sense to reset the stage in these cases. i can set that to be the default CTA here, and in a future PR where we have finer handling of login errors we can have more appropriate CTAs for each cases